Assets/Scripts/Projectile.cs

```diff
@@ -0,0 +1,38 @@
+﻿using System.Collections;
+using UnityEngine;
+
+public class Projectile : MonoBehaviour
+{
+    public int damagePoints;
+    public bool destructible;
+    public int healthPoints;
+    public bool timeLimit;
+    public int lifetime;
+
+    public int manaCost;
+    // Start is called before the first frame update
+    private void Start()
+    {
+        if (timeLimit){
+            StartCoroutine(deathCoroutine());
+        }
+    }
+
+    // Update is called once per frame
+    private void Update()
+    {
+        
+    }
+
+    private void OnCollisionEnter2D(Collision2D collision){
+        if(destructible){
+            Destroy(gameObject);
+        }
+    }
```

Comment:
I would try to make it so that the projectile object deals damage when it hits a damageable object instead of relying on the objects to check for collisions (look into using an interface).